src/config_parse.c

```diff
@@ -217,7 +217,7 @@ static int skip_bom(git_parse_ctx *parser)
 {
 	git_buf buf = GIT_BUF_INIT_CONST(parser->content, parser->content_len);
 	git_bom_t bom;
-	int bom_offset = git_buf_text_detect_bom(&bom, &buf, parser->content_len);
+	int bom_offset = git_buf_text_detect_bom(&bom, &buf, 0);
```

Comment:
I understand your confusion.  It appears that we never use `git_buf_text_detect_bom` to detect a BOM with a non-0 offset.

It's not clear to me why this was ever added in the first place - when would we try to detect a BOM in the midst of a buffer?

Whatever the reason, it sounds like we've refactored things to always look at the beginning of the buffer.  Should we drop this confusing third arg?

Reply:
I think so. I even wonder if there is any usecase with a mid-buffer-BOM, considering they are only allowed at the beginning of a file.